clang/lib/3C/Utils.cpp

```diff
@@ -322,8 +322,7 @@ bool hasVoidType(clang::ValueDecl *D) { return isTypeHasVoid(D->getType()); }
 //  return D->isPointerType() == S->isPointerType();
 //}
 
-static bool castCheck(clang::QualType DstType, clang::QualType SrcType,
-                      bool AllowVoidCast) {
+static bool castCheck(clang::QualType DstType, clang::QualType SrcType) {
```

Comment:
Because we're now considering casts from `void*` as wild, so we never allow the void cast. Instead, we have exceptions for generic function returns and allocators outside of checked headers, which are essentially generic.